app/livechat/server/lib/routing/AutoSelection.js

```diff
@@ -20,11 +21,13 @@ class AutoSelection {
 	}
 
 	getNextAgent(department, ignoreAgentId) {
-		if (department) {
-			return LivechatDepartmentAgents.getNextAgentForDepartment(department, ignoreAgentId);
-		}
+		if (!settings.get('Livechat_kill_switch')) {
```

Comment:
### How about an early return here as well?

sample code for reference:

```
if (settings.get('Livechat_kill_switch')) {
	return;
}
if (department) {
	return LivechatDepartmentAgents.getNextAgentForDepartment(department, ignoreAgentId);
}
return Users.getNextAgent(ignoreAgentId);
```

Reply:
Optimized code blocks.